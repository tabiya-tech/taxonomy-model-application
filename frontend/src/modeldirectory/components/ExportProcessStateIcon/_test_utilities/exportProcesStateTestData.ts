import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import ExportProcessStateAPISpecs from "api-specifications/importProcessState";
import { getMockId } from "src/_test_utilities/mockMongoId";

export const getAllExportProcessStatePermutations = (): ModelInfoTypes.ExportProcessState[] => {
  // generate all permutations of ExportProcessState status, errored, exportErrors, exportWarnings
  const allStatuses = Object.values(ExportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  const allBooleans = [true, false];

  const allPermutations: ModelInfoTypes.ExportProcessState[] = [];
  allStatuses.forEach((status) => {
    allBooleans.forEach((errored) => {
      allBooleans.forEach((exportErrors) => {
        allBooleans.forEach((exportWarnings) => {
          allPermutations.push({
            id: getMockId(allPermutations.length + 1),
            status,
            result: {
              errored,
              exportErrors: exportErrors,
              exportWarnings: exportWarnings,
            },
            downloadUrl: "https://example.com",
            timestamp: new Date(),
          });
        });
      });
    });
  });
  return allPermutations;
};
