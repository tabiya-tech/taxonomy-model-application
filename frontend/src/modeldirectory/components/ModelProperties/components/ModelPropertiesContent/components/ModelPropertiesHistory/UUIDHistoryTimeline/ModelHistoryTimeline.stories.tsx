import type { Meta, StoryObj } from "@storybook/react";
import ModelHistoryTimeline from "./ModelHistoryTimeline";
import {
  getFakeUUIDHistoryDetailsArray,
  getFakeUUIDHistoryDetailsArrayWithUnresolvedUUIDS
} from "./_test_utilities/mockData";

const meta: Meta<typeof ModelHistoryTimeline> = {
  title:
    "ModelDirectory/ModelProperties/ModelPropertiesContentPanels/ModelPropertiesHistoryComponents/ModelHistoryTimeline",
  component: ModelHistoryTimeline,
  tags: ["autodocs"],
  args: {
    UUIDHistoryDetails: getFakeUUIDHistoryDetailsArray(1),
  },
};

type Story = StoryObj<typeof ModelHistoryTimeline>;

export const Shown: Story = {};
export const ShownWithMultiple: Story = {
  args: {
    UUIDHistoryDetails: getFakeUUIDHistoryDetailsArray(5),
  },
};
export const ShownWithUnresolvedUUIDS: Story = {
  args: {
    UUIDHistoryDetails: getFakeUUIDHistoryDetailsArrayWithUnresolvedUUIDS(5),
  },
};

export const ShownWithMixedResolvedAndUnresolvedUUIDS: Story = {
  args: {
    UUIDHistoryDetails: [...getFakeUUIDHistoryDetailsArrayWithUnresolvedUUIDS(3), ...getFakeUUIDHistoryDetailsArray(3)].sort(() => 0.5 - Math.random()),
  },
};

export default meta;
