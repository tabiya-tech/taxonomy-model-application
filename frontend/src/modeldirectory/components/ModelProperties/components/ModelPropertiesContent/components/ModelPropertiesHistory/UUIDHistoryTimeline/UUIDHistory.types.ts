export interface IUUIDHistoryDetails {
  id: string;
  UUID: string;
  name: string;
  version: string;
  localeShortCode: string;
}