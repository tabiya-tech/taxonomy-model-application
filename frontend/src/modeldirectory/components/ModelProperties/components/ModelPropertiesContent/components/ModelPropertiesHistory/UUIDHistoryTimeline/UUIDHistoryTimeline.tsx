import React from "react";
import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";

import {
  Timeline,
  TimelineConnector,
  TimelineContent,
  TimelineDot,
  TimelineItem,
  timelineItemClasses,
  TimelineSeparator,
} from "@mui/lab";
import { Typography } from "@mui/material";

interface UUIDHistoryTimelineProps {
  UUIDHistoryDetails: ModelInfoTypes.UUIDHistory[];
}

const uniqueId = "48d564f1-fe4b-4c2f-bfc4-023cde28397f";
export const DATA_TEST_ID = {
  UUID_HISTORY_TIMELINE: `uuid-history-timeline-${uniqueId}`,
  UUID_HISTORY_TIMELINE_ITEM: `uuid-history-timeline-item-${uniqueId}`,
  UUID_HISTORY_TIMELINE_SEPARATOR: `uuid-history-timeline-separator-${uniqueId}`,
  UUID_HISTORY_TIMELINE_DOT: `uuid-history-timeline-dot-${uniqueId}`,
  UUID_HISTORY_TIMELINE_CONNECTOR: `uuid-history-timeline-connector-${uniqueId}`,
  UUID_HISTORY_TIMELINE_CONTENT: `uuid-history-timeline-content-${uniqueId}`,
};

const UUIDHistoryTimeline: React.FC<UUIDHistoryTimelineProps> = (props) => {
  return (
    <Timeline
      data-testid={DATA_TEST_ID.UUID_HISTORY_TIMELINE}
      sx={{
        [`& .${timelineItemClasses.root}:before`]: {
          flex: 0,
          padding: 0,
        },
      }}
    >
      {props.UUIDHistoryDetails.map((UUIDHistory) => (
        <TimelineItem
          data-testid={DATA_TEST_ID.UUID_HISTORY_TIMELINE_ITEM}
          key={UUIDHistory.UUID}
          sx={{
            minHeight: 0,
          }}
        >
          <TimelineSeparator data-testid={DATA_TEST_ID.UUID_HISTORY_TIMELINE_SEPARATOR}>
            <TimelineDot data-testid={DATA_TEST_ID.UUID_HISTORY_TIMELINE_DOT} color="primary" sx={{ margin: 0 }} />
            <TimelineConnector
              data-testid={DATA_TEST_ID.UUID_HISTORY_TIMELINE_CONNECTOR}
              sx={{ backgroundColor: (theme) => theme.palette.primary.main }}
            />
          </TimelineSeparator>
          <TimelineContent
            data-testid={DATA_TEST_ID.UUID_HISTORY_TIMELINE_CONTENT}
            sx={{ display: "flex", flexDirection: "column", justifyContent: "center" }}
          >
            <Typography variant="body1" sx={{ textAlign: "start" }}>
              {`${UUIDHistory.name} ${UUIDHistory.version} (${UUIDHistory.localeShortCode})`}
            </Typography>
          </TimelineContent>
        </TimelineItem>
      ))}
    </Timeline>
  );
};

export default UUIDHistoryTimeline;
