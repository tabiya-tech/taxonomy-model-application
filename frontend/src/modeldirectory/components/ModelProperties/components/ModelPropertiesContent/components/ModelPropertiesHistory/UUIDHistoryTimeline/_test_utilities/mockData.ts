import { ModelInfoTypes } from "src//modelInfo/modelInfoTypes";

export function getOneFakeUUIDHistoryDetail(i: number): ModelInfoTypes.ModelHistory {
  return {
    UUID: "e26aca13-aa00-4346-9e83-edaf59314378" + i,
    id: "5e12f537f079fbdcb4e61bf" + i,
    localeShortCode: "ZA",
    name: "Model" + i,
    version: "V1.1." + i,
  };
}

export function getFakeUUIDHistoryDetailsArray(length: number): ModelInfoTypes.ModelHistory[] {
  return Array.from({ length }, (_, i) => getOneFakeUUIDHistoryDetail(i));
}

export function getFakeUUIDHistoryDetailsArrayWithUnresolvedUUIDS(length: number): ModelInfoTypes.ModelHistory[] {
  return Array.from({ length }, (_, i) => {
    const details = getOneFakeUUIDHistoryDetail(i);
    details.id = null;
    details.name = null;
    details.version = null;
    details.localeShortCode = null;
    return details;
  });
}
