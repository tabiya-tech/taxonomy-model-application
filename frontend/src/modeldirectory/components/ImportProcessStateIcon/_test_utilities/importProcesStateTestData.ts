import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import ImportProcessStateAPISpecs from "api-specifications/importProcessState";
import { getMockId } from "src/_test_utilities/mockMongoId";

export const getAllImportProcessStatePermutations = (): ModelInfoTypes.ImportProcessState[] => {
  // generate all permutations of ImportProcessState status, errored, parsingErrors, parsingWarnings
  const allStatuses = Object.values(ImportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  const allBooleans = [true, false];

  const allPermutations: ModelInfoTypes.ImportProcessState[] = [];
  allStatuses.forEach((status) => {
    allBooleans.forEach((errored) => {
      allBooleans.forEach((parsingErrors) => {
        allBooleans.forEach((parsingWarnings) => {
          allPermutations.push({
            id: getMockId(allPermutations.length + 1),
            status,
            result: {
              errored,
              parsingErrors: parsingErrors,
              parsingWarnings: parsingWarnings,
            },
            createdAt: new Date(allPermutations.length),
            updatedAt: new Date(allPermutations.length),
          });
        });
      });
    });
  });
  return allPermutations;
};
