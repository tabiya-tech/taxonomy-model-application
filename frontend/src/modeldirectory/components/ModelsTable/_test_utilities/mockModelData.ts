import ModelInfoAPISpecs from "api-specifications/modelInfo";
import LocaleAPISpecs from "api-specifications/locale";
import ImportProcessStateAPISpecs from "api-specifications/importProcessState";
import ExportProcessStateAPISpecs from "api-specifications/exportProcessState";
import { ImportProcessStateEnums } from "api-specifications/importProcessState/enums";

import { v4 as uuidv4 } from "uuid";

import { ModelInfoTypes } from "src/modelInfo/modelInfoTypes";
import { getMockId } from "src/_test_utilities/mockMongoId";
import { getRandomLorem, getRandomString, getTestString } from "src/_test_utilities/specialCharacters";
import { faker } from "@faker-js/faker";
import { CELL_MAX_LENGTH } from "src/modeldirectory/components/ModelsTable/ModelsTable";

export function getOneFakeModel(id?: number): ModelInfoTypes.ModelInfo {
  const model = getArrayOfFakeModels(1)[0];
  model.id = getMockId(id ?? 1);
  return model;
}

export function getArrayOfFakeModels(count: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < count; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      UUIDHistory: [uuidv4()],
      name: `${i + 1}/${count} - ${getRandomLorem(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH)}`.substring(
        0,
        ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH
      ),
      locale: {
        UUID: uuidv4(),
        name: faker.location.country().substring(0, ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: faker.location
          .countryCode("alpha-3")
          .substring(0, LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH),
      },
      description:
        i % 2 === 0
          ? getRandomLorem(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH)
          : getRandomLorem(CELL_MAX_LENGTH / 2), // 50% chance of long description
      released: i % 2 === 0, // 50% chance of released
      releaseNotes: faker.lorem.text().substring(0, ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: faker.system.semver().substring(0, ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      exportProcessState: [getOneFakeExportProcessState(i)],
      importProcessState: getOneFakeImportProcessState(i)
    };
    models.push(model);
  }
  return models;
}

export function getArrayOfFakeModelsMaxLength(count: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < count; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      UUIDHistory: [uuidv4()],
      name: `${i + 1}/${count} - ${getRandomLorem(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH)}`.substring(
        0,
        ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH
      ),
      locale: {
        UUID: uuidv4(),
        name: getRandomLorem(LocaleAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: getRandomLorem(LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH),
      },
      description: getRandomLorem(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH),
      released: i % 2 === 0, // 50% chance of released
      releaseNotes: getRandomLorem(ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: getRandomLorem(ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      exportProcessState: [getOneFakeExportProcessState(i)],
      importProcessState: getOneFakeImportProcessState(i)
    };
    models.push(model);
  }
  return models;
}

export function getOneRandomModelMaxLength(): ModelInfoTypes.ModelInfo {
  return getArrayOfRandomModelsMaxLength(1)[0];
}

export function getArrayOfRandomModelsMaxLength(number: number): ModelInfoTypes.ModelInfo[] {
  const models: ModelInfoTypes.ModelInfo[] = [];
  for (let i = 0; i < number; i++) {
    const model: ModelInfoTypes.ModelInfo = {
      id: getMockId(i),
      UUID: uuidv4(),
      UUIDHistory: [uuidv4()],
      name: getRandomString(ModelInfoAPISpecs.Constants.NAME_MAX_LENGTH),
      locale: {
        UUID: uuidv4(),
        name: getRandomString(LocaleAPISpecs.Constants.NAME_MAX_LENGTH),
        shortCode: getTestString(LocaleAPISpecs.Constants.LOCALE_SHORTCODE_MAX_LENGTH),
      },
      description: getTestString(ModelInfoAPISpecs.Constants.DESCRIPTION_MAX_LENGTH),
      released: i % 2 === 0,
      releaseNotes: getTestString(ModelInfoAPISpecs.Constants.RELEASE_NOTES_MAX_LENGTH),
      version: getTestString(ModelInfoAPISpecs.Constants.VERSION_MAX_LENGTH),
      createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
      updatedAt: new Date(),
      path: faker.internet.url(),
      tabiyaPath: faker.internet.url(),
      exportProcessState: [getOneFakeExportProcessState(i)],
      importProcessState: getOneFakeImportProcessState(i)
    };
    models.push(model);
  }
  return models;
}

export const fakeModel: ModelInfoTypes.ModelInfo = {
  UUID: "8d914eab-6f7d-4183-accc-a09f99887b39",
  UUIDHistory: ["8d914eab-6f7d-4183-accc-a09f99887b39"],
  description: "aw j zt   agrkasl dy ogtimpsauwumu l utrovthao syertm beawpxluhyudgzbbm",
  id: "000000000000000000000001",
  importProcessState: {
    id: "000000000000000000000001",
    result: {
      errored: true,
      parsingErrors: true,
      parsingWarnings: true,
    },
    status: ImportProcessStateEnums.Status.PENDING,
    createdAt: new Date("2023-10-18T17:35:10.571Z"),
    updatedAt: new Date("2023-10-18T17:35:12.571Z"),
  },
  exportProcessState: [
    {
      downloadUrl: "https://www.example.com/",
      id: "000000000000000000000001",
      result: {
        errored: true,
        exportErrors: true,
        exportWarnings: true,
      },
      status: ExportProcessStateAPISpecs.Enums.Status.PENDING,
      timestamp: new Date("2023-10-18T17:35:10.571Z"),
      createdAt: new Date("2023-10-18T17:35:10.571Z"),
      updatedAt: new Date("2023-10-18T17:35:12.571Z"),
    },
  ],
  locale: {
    UUID: "7c4ea00b-fbea-4a31-9c04-009c813f5f34",
    name: "Palestine",
    shortCode: "ETH",
  },
  name: "1/1 - fysmros tcgcnjbbrrev",
  path: "https://unwelcome-editorial.net/",
  releaseNotes: "Perferendis modi impedit necessitatibus a",
  released: true,
  tabiyaPath: "https://silver-vulture.biz",
  createdAt: new Date("2023-10-18T17:35:10.571Z"),
  updatedAt: new Date("2022-10-18T17:35:10.571Z"),
  version: "0.2.2",
};

export function getRandomImportStatus(id: number) {
  const allStatuses = Object.values(ImportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  return allStatuses[id % allStatuses.length];
}

export function getRandomExportStatus(id: number) {
  const allStatuses = Object.values(ExportProcessStateAPISpecs.Enums.Status); // Assuming it's an enum with string values
  return allStatuses[id % allStatuses.length];
}

export function getOneFakeExportProcessState(i: number): ModelInfoTypes.ExportProcessState {
  return {
    id: getMockId(10000 + i),
    status: getRandomExportStatus(i),
    result: {
      errored: i % 2 === 0,
      exportErrors: i % 2 === 0,
      exportWarnings: i % 2 === 0,
    },
    downloadUrl: faker.internet.url(),
    timestamp: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
    createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
    updatedAt: new Date(),
  };
}

export function getOneFakeImportProcessState(i: number): ModelInfoTypes.ImportProcessState {
  return {
    id: getMockId(10000 + i),
    status: getRandomImportStatus(i),
    result: {
      errored: i % 2 === 0,
      parsingErrors: i % 2 === 0,
      parsingWarnings: i % 2 === 0,
    },
    createdAt: new Date(Date.now() - i * 1000 * 60 * 60 * 24),
    updatedAt: new Date(),
  };
}
