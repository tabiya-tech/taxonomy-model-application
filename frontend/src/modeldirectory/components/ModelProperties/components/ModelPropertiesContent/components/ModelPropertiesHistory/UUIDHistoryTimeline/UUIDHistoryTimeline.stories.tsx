import type { Meta, StoryObj } from "@storybook/react";
import UUIDHistoryTimeline from "./UUIDHistoryTimeline";
import { getFakeUUIDHistoryDetailsArray } from "./_test_utilities/mockData";

const meta: Meta<typeof UUIDHistoryTimeline> = {
  title:
    "ModelDirectory/ModelProperties/ModelPropertiesContentPanels/ModelPropertiesHistoryComponents/UUIDHistoryTimeline",
  component: UUIDHistoryTimeline,
  tags: ["autodocs"],
  args: {
    UUIDHistoryDetails: getFakeUUIDHistoryDetailsArray(5),
  },
};

type Story = StoryObj<typeof UUIDHistoryTimeline>;

export const Shown: Story = {};

export default meta;
