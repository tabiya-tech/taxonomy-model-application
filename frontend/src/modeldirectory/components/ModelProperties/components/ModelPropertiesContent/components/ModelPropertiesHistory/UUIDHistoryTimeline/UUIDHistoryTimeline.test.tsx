// mute the console
import "src/_test_utilities/consoleMock";

import { IUUIDHistoryDetails } from "./UUIDHistory.types";
import { render, screen, within } from "@testing-library/react";
import UUIDHistoryTimeline, { DATA_TEST_ID } from "./UUIDHistoryTimeline";
import {getFakeUUIDHistoryDetailsArray} from "./_test_utilities/mockData";

describe("UUIDHistoryTimeline", () => {
  test("renders correctly", () => {
    // GIVEN a UUIDHistoryDetails array
    const givenUUIDHistoryDetails: IUUIDHistoryDetails[] = getFakeUUIDHistoryDetailsArray(5);

    // WHEN the UUIDHistoryTimeline component is rendered with the given UUIDHistoryDetails
    render(<UUIDHistoryTimeline UUIDHistoryDetails={givenUUIDHistoryDetails} />);

    // THEN expect no errors or warnings to have occurred
    expect(console.error).not.toHaveBeenCalled();
    expect(console.warn).not.toHaveBeenCalled();
    // AND the timeline to be shown
    const timelineElement = screen.getByTestId(DATA_TEST_ID.UUID_HISTORY_TIMELINE);
    expect(timelineElement).toBeInTheDocument();
    // AND for each UUIDHistoryDetails
    givenUUIDHistoryDetails.forEach((uuidHistoryDetails, index) => {
      // the timeline item to be shown,
      const timelineItemElement = within(timelineElement).getAllByTestId(DATA_TEST_ID.UUID_HISTORY_TIMELINE_ITEM)[index];
      expect(timelineItemElement).toBeInTheDocument();
      // the timeline separator to be shown in the item,
      const timelineSeparatorElement = within(timelineItemElement).getByTestId(
        DATA_TEST_ID.UUID_HISTORY_TIMELINE_SEPARATOR
      );
      expect(timelineSeparatorElement).toBeInTheDocument();
      // the timeline dot to be shown in the separator,
      const timelineDotElement = within(timelineSeparatorElement).getByTestId(DATA_TEST_ID.UUID_HISTORY_TIMELINE_DOT);
      expect(timelineDotElement).toBeInTheDocument();
      // the timeline connector to be shown in the separator,
      const timelineConnectorElement = within(timelineSeparatorElement).getByTestId(
        DATA_TEST_ID.UUID_HISTORY_TIMELINE_CONNECTOR
      );
      expect(timelineConnectorElement).toBeInTheDocument();
      // AND the timeline content to be shown in the item with the text formatted as: name version (localeShortCode)
      const timelineContentElement = within(timelineItemElement).getByTestId(
        DATA_TEST_ID.UUID_HISTORY_TIMELINE_CONTENT
      );
      expect(timelineContentElement).toBeInTheDocument();
      expect(timelineContentElement.textContent).toBe(
        `${uuidHistoryDetails.name} ${uuidHistoryDetails.version} (${uuidHistoryDetails.localeShortCode})`
      );
    });
    // AND the snapshot to match
    expect(timelineElement).toMatchSnapshot();
  });
});
