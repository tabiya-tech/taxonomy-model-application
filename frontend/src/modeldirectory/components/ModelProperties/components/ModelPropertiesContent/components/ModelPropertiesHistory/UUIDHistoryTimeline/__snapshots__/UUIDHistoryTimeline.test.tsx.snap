// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UUIDHistoryTimeline renders correctly 1`] = `
<ul
  class="MuiTimeline-root MuiTimeline-positionRight css-6l1m71-MuiTimeline-root"
  data-testid="uuid-history-timeline-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
>
  <li
    class="MuiTimelineItem-root MuiTimelineItem-positionRight MuiTimelineItem-missingOppositeContent css-1ak9pa5-MuiTimelineItem-root"
    data-testid="uuid-history-timeline-item-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
  >
    <div
      class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
      data-testid="uuid-history-timeline-separator-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
    >
      <span
        class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledPrimary css-1qwwwv0-MuiTimelineDot-root"
        data-testid="uuid-history-timeline-dot-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
      />
      <span
        class="MuiTimelineConnector-root css-iplxxw-MuiTimelineConnector-root"
        data-testid="uuid-history-timeline-connector-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
      />
    </div>
    <div
      class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionRight css-1oky1fk-MuiTypography-root-MuiTimelineContent-root"
      data-testid="uuid-history-timeline-content-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1v2d8t3-MuiTypography-root"
      >
        Name0 Version0 (ABC0)
      </p>
    </div>
  </li>
  <li
    class="MuiTimelineItem-root MuiTimelineItem-positionRight MuiTimelineItem-missingOppositeContent css-1ak9pa5-MuiTimelineItem-root"
    data-testid="uuid-history-timeline-item-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
  >
    <div
      class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
      data-testid="uuid-history-timeline-separator-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
    >
      <span
        class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledPrimary css-1qwwwv0-MuiTimelineDot-root"
        data-testid="uuid-history-timeline-dot-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
      />
      <span
        class="MuiTimelineConnector-root css-iplxxw-MuiTimelineConnector-root"
        data-testid="uuid-history-timeline-connector-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
      />
    </div>
    <div
      class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionRight css-1oky1fk-MuiTypography-root-MuiTimelineContent-root"
      data-testid="uuid-history-timeline-content-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1v2d8t3-MuiTypography-root"
      >
        Name1 Version1 (ABC1)
      </p>
    </div>
  </li>
  <li
    class="MuiTimelineItem-root MuiTimelineItem-positionRight MuiTimelineItem-missingOppositeContent css-1ak9pa5-MuiTimelineItem-root"
    data-testid="uuid-history-timeline-item-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
  >
    <div
      class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
      data-testid="uuid-history-timeline-separator-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
    >
      <span
        class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledPrimary css-1qwwwv0-MuiTimelineDot-root"
        data-testid="uuid-history-timeline-dot-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
      />
      <span
        class="MuiTimelineConnector-root css-iplxxw-MuiTimelineConnector-root"
        data-testid="uuid-history-timeline-connector-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
      />
    </div>
    <div
      class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionRight css-1oky1fk-MuiTypography-root-MuiTimelineContent-root"
      data-testid="uuid-history-timeline-content-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1v2d8t3-MuiTypography-root"
      >
        Name2 Version2 (ABC2)
      </p>
    </div>
  </li>
  <li
    class="MuiTimelineItem-root MuiTimelineItem-positionRight MuiTimelineItem-missingOppositeContent css-1ak9pa5-MuiTimelineItem-root"
    data-testid="uuid-history-timeline-item-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
  >
    <div
      class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
      data-testid="uuid-history-timeline-separator-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
    >
      <span
        class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledPrimary css-1qwwwv0-MuiTimelineDot-root"
        data-testid="uuid-history-timeline-dot-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
      />
      <span
        class="MuiTimelineConnector-root css-iplxxw-MuiTimelineConnector-root"
        data-testid="uuid-history-timeline-connector-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
      />
    </div>
    <div
      class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionRight css-1oky1fk-MuiTypography-root-MuiTimelineContent-root"
      data-testid="uuid-history-timeline-content-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1v2d8t3-MuiTypography-root"
      >
        Name3 Version3 (ABC3)
      </p>
    </div>
  </li>
  <li
    class="MuiTimelineItem-root MuiTimelineItem-positionRight MuiTimelineItem-missingOppositeContent css-1ak9pa5-MuiTimelineItem-root"
    data-testid="uuid-history-timeline-item-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
  >
    <div
      class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
      data-testid="uuid-history-timeline-separator-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
    >
      <span
        class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledPrimary css-1qwwwv0-MuiTimelineDot-root"
        data-testid="uuid-history-timeline-dot-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
      />
      <span
        class="MuiTimelineConnector-root css-iplxxw-MuiTimelineConnector-root"
        data-testid="uuid-history-timeline-connector-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
      />
    </div>
    <div
      class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionRight css-1oky1fk-MuiTypography-root-MuiTimelineContent-root"
      data-testid="uuid-history-timeline-content-48d564f1-fe4b-4c2f-bfc4-023cde28397f"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1v2d8t3-MuiTypography-root"
      >
        Name4 Version4 (ABC4)
      </p>
    </div>
  </li>
</ul>
`;
